### PR TITLE
Migrate integration tests for backups and restores from the k8s charm

### PR DESCRIPTION
**Pull request by shayancanonical**
[dpe-1275](https://warthogs.atlassian.net/browse/DPE-1275)

## Issue
- We have yet to migrate the integration tests for backups and restores from the k8s charm

## Solution
- Migrate tests

### CODE REVIEW

**Review by paulomach — APPROVED** [review]
lgtm

**Review by taurus-forever — APPROVED** [review]

File: tests/integration/test_backups.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        )
+        select_values_sql = [f"SELECT id FROM `{DATABASE_NAME}`.`{TABLE_NAME}`"]
+
+        values = await execute_queries_on_unit(
+            mysql_unit_address,
+            server_config_credentials["username"],
+            server_config_credentials["password"],
+            select_values_sql,
+        )
+        assert values == [value_before_backup]
```
Comment (delgod) [inline tests/integration/test_backups.py:251]: please broke data before run restoration for another cloud
```
    logger.info("Inserting value after backup")
    value_after_backup = await insert_data_into_mysql_and_validate_replication(
        ops_test,
        DATABASE_NAME,
        TABLE_NAME,
    )
```

File: tests/integration/test_backups.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+        )
+        select_values_sql = [f"SELECT id FROM `{DATABASE_NAME}`.`{TABLE_NAME}`"]
+
+        values = await execute_queries_on_unit(
+            primary_unit_address,
+            server_config_credentials["username"],
+            server_config_credentials["password"],
+            select_values_sql,
+        )
+        assert values == [value_before_backup]
```
Comment (delgod) [inline tests/integration/test_backups.py:332]: Please scale to 3 members after restoration, and check the data on all members

Comment (shayancanonical) [reply tests/integration/test_backups.py:370]: addressed in [f5d8606](https://github.com/canonical/mysql-operator/pull/145/commits/f5d8606cd2ce7237bf3051ba1f9b81ea6b2c8a7c)

File: tests/integration/test_backups.py (hunk 1 above)
Comment (shayancanonical) [reply tests/integration/test_backups.py:252]: addressed in [f5d8606](https://github.com/canonical/mysql-operator/pull/145/commits/f5d8606cd2ce7237bf3051ba1f9b81ea6b2c8a7c)

**Review by delgod — APPROVED** [review]

**Pull request merged**